recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.tmpl

```diff
@@ -702,17 +758,20 @@ def main(argv):
         The Device may become brick.
         Please try to upgrade firmware again!
         ====================================
-Hit any Key to Exit:
+Hit the Enter Key to Exit:
         '''
         # Rollback the firmware when the upgrade fails
         try:
-            print('Warning: Upgrade failed, rollback started')
-            updater.rollback_firmware()
+            print(BOLD + 'Warning: Upgrade failed, rollback started' + ENDC + '\n')
+            ret = updater.rollback_firmware()
+            if ErrorCode.ROLLBACK_FAILED.value == ret:
+                return ret
         except UpgradeError:
             print("Rollback failed")
 
-        input(intput_reminder)
-
+        the_input(False, intput_reminder)
+        return ErrorCode.INTERNAL_ERR.value
```

Comment:
We need to separate rollback or main binary flashing errors. Please recheck the flow of code for upgrading process for both normal and rollback parts.

Reply:
Also, any problem during the flashing process should return the fail return code.